doc-2596 (#2111)
harness/developer-hub: docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/kubernetes-cluster-connector-settings-reference.md

Looking for the How-to? See [Add a Kubernetes Cluster Connector](https://developer.harness.io/docs/platform/Connectors/Cloud-providers/add-a-kubernetes-cluster-connector).

## Video Summary

> Here's a 10min video that walks you through adding a Harness Kubernetes Cluster Connector and Harness Kubernetes Delegate. The Delegate is added to the target cluster and then the Kubernetes Cluster Connector uses the Delegate to connect to the cluster:

Reviewer: ```suggestion
Here's a 10min video that walks you through adding a Harness Kubernetes Cluster Connector and Harness Kubernetes delegate. The delegate is added to the target cluster and then the Kubernetes Cluster Connector uses the delegate to connect to the cluster:
```

For a detailed list of roles and policies, see [Harness Role-Based Access Control Overview](../../../4_Role-Based-Access-Control/1-rbac-in-harness.md#role).

### Harness CI Permission Requirements

If you are only using the Kubernetes Cluster Connector for Harness Continuous Integration (CI), you can use a reduced set of permissions.

> For Harness CI, the Delegate requires CRUD permissions on Secret and Pod.

Reviewer: ```suggestion
For Harness CI, the delegate requires CRUD permissions on Secret and Pod.
```

  name: sa-role  
  apiGroup: rbac.authorization.k8s.io
```
### Builds (CI)

A Kubernetes service account with CRUD permissions on Secret, Service, Pod, and PersistentVolumeClaim (PVC).

> For more information, see [User-Facing Roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles) from Kubernetes.

Reviewer: ```suggestion
For more information, go to [User-Facing Roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles) from Kubernetes.
```

## Credential Validation

When you click **Submit**, Harness uses the provided credentials to list controllers in the **default** namespace in order to validate the credentials. If validation fails, Harness does not save the Connector and the **Submit** fails.

If your cluster does not have a **default** namespace, or your credentials do not have permission in the **default** namespace, then you can check **Skip default namespace validation** to skip this check and saving your Connector settings.

You do not need to come back and uncheck **Skip default namespace validation**.

Later, when you define a target Infrastructure using this Connector, you will also specify a specific namespace. During deployment, Harness uses this namespace rather than the **default** namespace.

When Harness saves the Infrastructure it performs validation even if **Skip default namespace validation** was checked.

## Name

The unique name for this Connector.

## ID

See [Entity Identifier Reference](../../../20_References/entity-identifier-reference.md).

## Description

Text string.

## Tags

See [Tags Reference](../../../20_References/tags-reference.md).

## Cluster Details

> ### Manual or Use a Delegate

Reviewer: ```suggestion
### Manual or Use a delegate
```

> **Recommended:** Install and run the Harness Kubernetes Delegate in the target Kubernetes cluster, and then use the Kubernetes Cluster Connector to connect to that cluster using the Harness Kubernetes Delegate you installed. This is the easiest method to connect to a Kubernetes cluster.You can select to enter the authentication details of the target cluster or use the role associated with a Harness Delegate.

Reviewer: ```suggestion
**Recommended:** Install and run the Harness Kubernetes Delegate in the target Kubernetes cluster, and then use the Kubernetes Cluster Connector to connect to that cluster using the Harness Kubernetes delegate you installed. This is the easiest method to connect to a Kubernetes cluster.You can select to enter the authentication details of the target cluster or use the role associated with a Harness Delegate.
```

> **Recommended:** Install and run the Harness Kubernetes Delegate in the target Kubernetes cluster, and then use the Kubernetes Cluster Connector to connect to that cluster using the Harness Kubernetes Delegate you installed. This is the easiest method to connect to a Kubernetes cluster.You can select to enter the authentication details of the target cluster or use the role associated with a Harness Delegate.

Reviewer: ```suggestion
**Recommended:** Install and run the Harness Kubernetes delegate in the target Kubernetes cluster, and then use the Kubernetes Cluster Connector to connect to that cluster using the Harness Kubernetes delegate you installed. This is the easiest method to connect to a Kubernetes cluster.You can select to enter the authentication details of the target cluster or use the role associated with a Harness Delegate.
```

> When you select a Delegate, the Harness Delegate inherits the Kubernetes service account associated with the Delegate pod.

Reviewer: ```suggestion
When you select a delegate, the Harness Delegate inherits the Kubernetes service account associated with the delegate pod.
```

> The service account associated with the Delegate pod must have the Kubernetes `cluster-admin` role.

Reviewer: ```suggestion
The service account associated with the delegate pod must have the Kubernetes `cluster-admin` role.
```
